crates/driver/src/domain/competition/solution/interaction.rs

```diff
@@ -16,6 +16,32 @@ impl Interaction {
             Interaction::Liquidity(liquidity) => liquidity.internalize,
         }
     }
+
+    /// Returns the ERC20 allowances required for executing this interaction
+    /// onchain.
+    pub fn allowances(&self) -> Vec<eth::allowance::Required> {
+        match self {
+            Interaction::Custom(interaction) => interaction.allowances.clone(),
+            Interaction::Liquidity(interaction) => {
+                let address = match &interaction.liquidity.kind {
+                    liquidity::Kind::UniswapV2(pool) => pool.router.into(),
+                    liquidity::Kind::UniswapV3(_) => todo!(),
+                    liquidity::Kind::BalancerV2Stable(_) => todo!(),
+                    liquidity::Kind::BalancerV2Weighted(_) => todo!(),
+                    liquidity::Kind::Swapr(_) => todo!(),
+                    liquidity::Kind::ZeroEx(_) => todo!(),
+                };
```

Comment:
I would probably move this block into the `Liquidity` type (e.g. `Liquidity::allowances(&self)`), not because it's better to have shorter methods but because extensively matching on the liquidity kind belongs there more than it does in the `solution` module.